### ISSUE TITLE
Column Block: Wrong view for nested content (block editor)

### ISSUE THREAD

**Issue by CreativeDive** [WordPress/gutenberg#17834]
Hey,

Currently, the view is wrong for some nested blocks within the column block, **only in case of a vertical alignment like top, middle or bottom is selected**.

I think the flexbox should be reworked.

**Block editor issue: Test view nested cover block**

![column-bug](https://user-images.githubusercontent.com/31563167/66372721-fd98c000-e9a6-11e9-9529-0bdb6111a78b.gif)

**Block editor issue: Test view nested group block**

![Oct-08-2019 08-45-17](https://user-images.githubusercontent.com/31563167/66373088-12298800-e9a8-11e9-8160-629d2bde1783.gif)

This only happens in the block editor. The frontend view is correct.

**Frontend view:**

<img width="1058" alt="Bildschirmfoto 2019-10-08 um 08 42 12" src="https://user-images.githubusercontent.com/31563167/66372907-8d3e6e80-e9a7-11e9-9b73-7c778066af65.png">

Tested with Gutenberg 6.6
Theme: Twenty Nineteen

**Comment by dickmoomaw** [comment]
Tested this on Gutenberg Version 6.8.0 with MacOS Chrome Version 78.0.3904.70.  Did not reproduce the problem.

**Comment by Dabalina** [comment]
I was able to recreate this issue once I closed the sidebar and editor screen became wider. Not only was I able to recreate this with the cover block but I was also able to recreate it with the default placement block.

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/4277676/68092323-88a69200-fe4f-11e9-8acd-a4a4c71a1f61.gif)

**Comment by dickmoomaw** [comment]
Is the problem that changing the vertical alignment also changes the horizontal alignment?  That became apparent when I followed Dabalina's action.

**Comment by CreativeDive** [comment]
This problem occurs if the content is smaller than the width of the container block. You can re-create the problem by using the cover or group block with a selected background. Just use text like "Hello World" for the inner content.

**Comment by gwwar** [comment]
If folks are still seeing this issue, would it be possible to help document the use case? It looks like there might be more setup required to reproduce this.

One thing that'd be pretty helpful is copying the example block markup into a comment on this issue. We can do this by clicking on the three dots in the top right, then click on "code editor".

<img width="317" alt="Screen Shot 2021-04-22 at 12 38 12 PM" src="https://user-images.githubusercontent.com/1270189/115775632-fefec300-a367-11eb-9fd7-ddeb449880e7.png">

**Comment by paaljoachim** [comment]
I tested the examples that @CreativeDive Martin mentioned.

Using WordPress 5.7. Twenty Twenty One. Gutenberg plugin 10.4.1

Adding a 2 column wide Column block.
Added a Cover block into the first. Containing an image and some text in the middle of the Cover block.
Second column contained text. I was able to nicely align top, middle and bottom the Cover block.

Adding a 2 column wide Column block.
1st column: a Group block with a paragraph block inside. Added color to background.
2nd column: text.
I was able to nicely align top, middle and bottom the Cover block.

I assume there is something going on that I was not able to test.
